pkg/sql/catalog/lease/storage.go

```diff
@@ -180,12 +187,25 @@ func (s storage) acquire(
 		log.VEventf(ctx, 2, "storage attempting to acquire lease %v@%v", desc, expiration)
 
 		ts := storedLeaseExpiration(expiration)
+		if session != nil {
+			sessionID = session.ID().UnsafeBytes()
+		}
 		lf := leaseFields{
 			regionPrefix: prefix,
 			descID:       desc.GetID(),
 			version:      desc.GetVersion(),
 			instanceID:   s.nodeIDContainer.SQLInstanceID(),
 			expiration:   ts,
+			sessionID:    sessionID,
+		}
+		// For session based leases, we will only insert if the version changes
+		// or if the current lease doesn't have a session ID.
+		if lastLease.sessionID != nil &&
```

Comment:
I'm thoroughly confused about this comment and block of code

Does current lease here mean `lastLease`? If so, shouldn't the check be `lastLeast.sessionID == nil`?

It also says "we will only insert if ..." but `insertLease` is always called. Is `insertLease` doing some extra logic or is there a missing return that will be present for session only mode?